pkg/controller/equality.go

```diff
@@ -93,6 +93,29 @@ func ServiceEqual(newSvc, oldSvc *corev1.Service) (bool, error) {
 	return false, nil
 }
 
+// ServiceAnnLabelEqual compares the new Services's annotation and lables with old Services
+func ServiceAnnLabelEqual(newSvc, oldSvc *corev1.Service) (bool, error) {
```

Comment:
Hi @wxiaomou, thanks for the refactoring work! I'm considering can we take one step further, take an additional `runtime.Object` parameter (which is the input for `m.deps.ServiceControl.UpdateService`), and a modifier function to sync cluster IP (`svc.Spec.ClusterIP = oldSvc.Spec.ClusterIP`) in the function signature? Then we can unify the service syncing logic for each component, and `ServiceAnnLabelEqual` becomes something like `SyncComponentService` semantically. What do you think?

Reply:
@hanlins refactored and add an unit test for SyncComponentService. Thanks for the review.

Reply:
@hanlins I didn't apply this refactor for tidb_manager and pump_manager since they are checking if the service is orphan, do you have any suggestion? We can bypass them or add the orphan check in SyncComponentService. WDYT

Reply:
I think it should be fine, we could leave that as a follow-up improvement to take care of the orphan-checking logic in a separate function.